src/components/Sidebars/Payback.js

```diff
@@ -92,6 +113,37 @@ const Payback = ({ cdp, reset }) => {
             }
           />
         </Grid>
+        {!allowanceExistsOnMount && (
+          <Grid gridTemplateColumns="2fr 1fr" gridColumnGap="s">
+            <Text.p t="body">{lang.action_sidebar.unlock_dai}</Text.p>
+            {activeSpinner ? (
+              <Box alignSelf="center" justifySelf="end">
+                <Loader size="20" />
+              </Box>
+            ) : (
+              <Toggle
+                active={hasAllowance}
+                onClick={async () => {
+                  try {
+                    if (!hasAllowance) {
+                      setActiveSpinner(true);
+                      await token.approveUnlimited(proxy);
```

Comment:
probably ok to do something like 
```
const daiToken = maker.getToken('MDAI');
daiToken.approveUnlimited(proxy);
```
rather than storing the object in state - what do you think?

Reply:
:+1: